lib/puppet/provider/package/gem.rb

```diff
@@ -15,7 +15,52 @@
 
   has_feature :versionable, :install_options, :uninstall_options
 
-  commands :gemcmd => "gem"
+  def self.which_windows_gemcmd(bin)
```

Comment:
If we're going to build our own `which` I would suggest we drop it in our custom file system support.  A more complete cross-platform implementation is at https://gist.github.com/steakknife/88b6c3837a5e90a08296 FWIW